pydantic/functional_serializers.py

```diff
@@ -32,11 +51,32 @@ def __get_pydantic_core_schema__(self, source_type: Any, handler: GetCoreSchemaH
 
 @slots_dataclass(frozen=True)
 class WrapSerializer:
+    """
+    This class is used to wrap a serializer function and bind it with requested parameters for later use in
+    schema building.
+
+    Attributes:
+        func (core_schema.WrapSerializerFunction): The function to be wrapped.
+        json_return_type (Optional[core_schema.JsonReturnTypes]): The JSON return type annotation of the function.
+        when_used (Literal['always', 'unless-none', 'json', 'json-unless-none']): Determines the way the serialization
+            will go. This governs whether and when your function's return value will be serialized.
```

Comment:
@adriangb I think you might be able to improve on this description, and I think it's worth making it clear since I think this is something people will _actually_ benefit from the docs for, unlike some of the things in this PR (such as most of the descriptions of the GenerateJsonSchema methods).

Reply:
I think it might be enough to just change:
```suggestion
            will go. This governs whether and when this function will be used during serialization.
```